src/boot/efi/stub.c

```diff
@@ -469,6 +539,7 @@ static EFI_STATUS run(EFI_HANDLE image) {
         err = cmdline_append_and_measure_addons(
                         image,
                         loaded_image,
+                        true,
```

Comment:
This is a bit strange - loading from the protocol is grafted onto one of the filesystem load calls. Instead, things should be refactored so that they are 3 individual calls, so that they are nicely separated and individual.

Reply:
With your new PR, I moved the stuff inside the `load_addons` call and it calls `load_addons_from_efi`, let me know if you still think I should split it in the main entrypoint and make it more visible.